### ISSUE TITLE
Import from old Plugin not work

### ISSUE THREAD

**Issue by xstable** [hicknhack-software/redmine_hourglass#50]
I've installed Hourglass and try to import Data from Redmine Time Tracker Plugin.

I only got a no helping error-message: 

> Es gab einen Fehler, die Daten vom Redmine Time Tracker konnten nicht importiert werden. Haben sie diese vielleicht bereits importiert?

Means: An error occured. Data from Redmine Time Tracker couldn't imported. Do you have still import them.

So I doesn't have import them yet. How can I fix this?

**Comment by xstable** [comment]
Ok, I've seen there are some time-data in a List of Hourglass, But they are all unasigned (not assigned to an project or an task).

Even If I open an Ticket, track some time and stop the time-tracker.... this tracked time is not assigned to an Projekt or Task.
Maybe all this cause of the same issue... even like my other bug of the App (https://github.com/hicknhack-software/redmine_hourglass/issues/51)

**Comment by jume-dev** [comment]
Please check if the plugin is enabled on your project and you have set the appropriate access rights.

**Comment by xstable** [comment]
Yes, works like a charm. Thanks a lot.

**Comment by xstable** [comment]
I was a little bit to fast. Now the Project-Pulldown works. But the Imported data are not assigned to an project.
Should I maybe wipe data of all houreglas-Database-Tables and do the import again (now that I've added the module to each projects)?

**Comment by jume-dev** [comment]
yes

**Comment by xstable** [comment]
Ok, I did this. But while importing I got again the message:

> Es gab einen Fehler, die Daten vom Redmine Time Tracker konnten nicht importiert werden. Haben sie diese vielleicht bereits importiert?

Means: An error occured. Data from Redmine Time Tracker couldn't imported. Do you have still import them.

If I have a look at time_bookings, there appear only timebookings between 16.06.2014 and 23.10.2013.

Maybe cause of an Timeout the import is interrupted? 
Is there a way to manually import (while checking till which date was import and adding the following one?)

**Comment by jume-dev** [comment]
Mhm, seems like we need to increase reporting for the import feature. Could you please check the logs, it should show you which records failed.

Btw how many entries do you have that you want to import?

At the moment there is now other way to import built in. If you are familiar with ruby, you can try fiddle with the code in lib/hourglass/redmine_time_tracker_import.rb

But iirc, you can just run the import over and over again, it should only import the ones not yet imported

**Comment by xstable** [comment]
Currently, 209 entries are imported, I have 1273 entries.

I'm only familiar with php, js, but no ruby, Still looked inside this plugin, but better touching nothing of it ^^

Have now clear the hourglass-db's and start over the import. I've attached the logs for this.
[logfile_hourglass.log](https://github.com/hicknhack-software/redmine_hourglass/files/1355432/logfile_hourglass.log)


To run the import over and over didn't work. Run it surveral times, but still have 209 Entries.

**Comment by jume-dev** [comment]
mhm, that's all? there needs to be some sort of actual error otherwise the import wouldn't stop.

**Comment by xstable** [comment]
thats the whole logfile.
Maybe I could send you timetracker db export and you try it at your development-system?

**Comment by jume-dev** [comment]
Sadly i can't provide more support than i did for now, cause i am really busy with another project for a customer.
2 solutions, as said in the other issue think about contacting us officially via our website or debug and try to find the error yourself.
If i saw correctly, you have rubymine? Then place a breakpoint in hourglass_import_controller on Line 7 and inspect on the exception.

**Comment by xstable** [comment]
Don't know what rubymine is.
Ruby is a deep black whole for me. Know nothing about it. 
A pitty, but seems I have to use the old timetracker-plugin (if it still work). Thanks for trying to help.

**Comment by jume-dev** [comment]
Well, you could use them in parallel annd use the new one only for new projects but that's probably a big hassle.
Sorry, i couldn't help you more

**Comment by xstable** [comment]
Whats about if you told me in wich way the migration is done. 
I could manually via SQL-Statement parse the old  entries into the new database.

**Comment by xstable** [comment]
Seems there where some entries in timetraker_bookings table, which breaks the script.
I've delete those, following the las dataset which was impored and start over the import. This fill all entries at least.

**Comment by danielriesner** [comment]
I probably came across the same bug. The import task fails for a time_bookings record that has a shorter duration then the associated time_logs record. I opened a [new issue](https://github.com/hicknhack-software/redmine_hourglass/issues/71) for it and added more details from the rake error log.
